lib/azure/armrest/armrest_service.rb

```diff
@@ -35,13 +36,20 @@ def token
         def fetch_token
           token_url = Azure::Armrest::AUTHORITY + tenant_id + "/oauth2/token"
 
-          response = JSON.parse(ArmrestService.rest_post(
-            token_url,
-            :grant_type    => grant_type,
-            :client_id     => client_id,
-            :client_secret => client_key,
-            :resource      => Azure::Armrest::RESOURCE
-          ))
+          response = JSON.parse(
```

Comment:
why not call `ArmrestService.rest_post` since you already refactored `rest_post` method?